labelbox/data/serialization/labelbox_v1/label.py

```diff
@@ -174,44 +171,49 @@ def from_common(cls, label: Label):
                          external_id=label.data.external_id,
                          **label.extra)
 
-    def _infer_media_type(self):
-        # Video annotations are formatted differently from text and images
-        # So we only need to differentiate those two
+    def _data_row_to_common(self) -> Union[ImageData, TextData, VideoData]:
+        # Use data row information to construct the appropriate annotatin type
         data_row_info = {
+            'url' if self._is_url() else 'text': self.row_data,
             'external_id': self.external_id,
             'uid': self.data_row_id
         }
 
+        self.media_type = self.media_type or self._infer_media_type()
+        media_mapping = {
+            'text': TextData,
+            'image': ImageData,
+            'video': VideoData
+        }
+        if self.media_type not in media_mapping:
+            raise ValueError(
+                f"Annotation types are only supported for {list(media_mapping)} media types."
+                f" Found {self.media_type}.")
+        return media_mapping[self.media_type](**data_row_info)
+
+    def _infer_media_type(self) -> str:
+        # Determines the data row type based on the label content
+        if isinstance(self.label, list):
+            return 'video'
         if self._has_text_annotations():
-            # If it has text annotations then it must be text
-            if self._is_url():
-                return TextData(url=self.row_data, **data_row_info)
-            else:
-                return TextData(text=self.row_data, **data_row_info)
+            return 'text'
         elif self._has_object_annotations():
-            # If it has object annotations and none are text annotations then it must be an image
-            if self._is_url():
-                return ImageData(url=self.row_data, **data_row_info)
-            else:
-                return ImageData(text=self.row_data, **data_row_info)
+            return 'image'
         else:
-            # no annotations to infer data type from.
-            # Use information from the row_data format if possible.
             if self._row_contains((".jpg", ".png", ".jpeg")) and self._is_url():
```

Comment:
can this be an or statement with line 200? `elif self._has_object_annotations():`

Reply:
We could but I think this is more readable